x/bsonx/bsoncore/array.go

```diff
@@ -82,34 +83,54 @@ func (a Array) DebugString() string {
 // String outputs an ExtendedJSON version of Array. If the Array is not valid, this method
 // returns an empty string.
 func (a Array) String() string {
-	if len(a) < 5 {
+	return a.StringN(math.MaxInt)
+}
+
+// StringN stringifies an array upto N bytes
+func (a Array) StringN(n int) string {
+	if lens, _, _ := ReadLength(a); lens < 5 || n <= 0 {
 		return ""
 	}
+
 	var buf strings.Builder
 	buf.WriteByte('[')
 
 	length, rem, _ := ReadLength(a) // We know we have enough bytes to read the length
-
 	length -= 4
 
 	var elem Element
 	var ok bool
-	for length > 1 {
-		elem, rem, ok = ReadElement(rem)
-		length -= int32(len(elem))
-		if !ok {
-			return ""
+
+	if n > 0 {
+		for length > 1 {
+			elem, rem, ok = ReadElement(rem)
+
+			length -= int32(len(elem))
+			if !ok {
+				return ""
+			}
+
+			str := elem.Value().StringN(n - buf.Len())
+
+			buf.WriteString(str)
+
+			if buf.Len() == n {
+				return buf.String()
+			}
```

Comment:
Addition of this code, fixed a bug that was making truncation 11 instead of the length of 10 in one of the test cases.
It also improved the benchmarking speed of truncating massive arrays.
Before Change-
BenchmarkRawString/massive_arrays_StringN-10                 642       1860169 ns/op      632576 B/op        411 allocs/op

After Change-
BenchmarkRawString/massive_arrays_StringN-10              105404         10738 ns/op       15744 B/op        396 allocs/op